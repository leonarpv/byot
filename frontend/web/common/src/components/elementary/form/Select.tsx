import React from 'react';
import {makeStyles, Select as MuiSelect, SelectProps, Theme, WithStyles} from '@material-ui/core';
import {InputBase, Props as InputBaseProps} from './InputBase';
import classNames from 'classnames';

type Props = Omit<InputBaseProps<SelectProps>, 'component'> & Partial<WithStyles<typeof styles>>;

const styles = (theme: Theme) => ({
  select: {
    paddingLeft: theme.spacing(1 / 2),
    minWidth: 50,
    '&:focus': {
      backgroundColor: 'transparent',
    },
  },
});
const useStyles = makeStyles(styles);

export const Select: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <InputBase
      component={MuiSelect}
      {...props}
      classes={{select: classNames(styles.select, props.classes?.select)}}
    />
  );
};
