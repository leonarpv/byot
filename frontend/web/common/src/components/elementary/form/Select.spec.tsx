import * as React from 'react';
import {render} from '@testing-library/react';
import {Select} from './Select';

describe('<Select/>', () => {
  it('should render', () => {
    const {container} = render(<Select />);
    expect(container).toMatchSnapshot();
  });
});
