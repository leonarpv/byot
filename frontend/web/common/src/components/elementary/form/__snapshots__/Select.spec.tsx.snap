// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select/> should render 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-1 makeStyles-filledRoot-3 MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <div
        aria-haspopup="listbox"
        class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
        role="button"
        tabindex="0"
      >
        <span>
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <div
    class="MuiCollapse-container MuiCollapse-hidden"
    style="min-height: 0px;"
  >
    <div
      class="MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner"
      >
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          <span>
            ​
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
